service/internal/moduleinfo/moduleinfo.go

```diff
@@ -0,0 +1,15 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package moduleinfo // import "go.opentelemetry.io/collector/service/internal/moduleinfo"
+
+import "go.opentelemetry.io/collector/component"
+
+// ModuleInfo describes the go module for each component.
+type ModuleInfo struct {
+	Receiver  map[component.Type]string
+	Processor map[component.Type]string
+	Exporter  map[component.Type]string
+	Extension map[component.Type]string
+	Connector map[component.Type]string
```

Comment:
Thinking if we should use this moment at least to re-design a bit this:

```
type ModuleInfo struct {
	ModuleName string
}

type ModuleInfos struct {
	Receiver  map[component.Type]ModuleInfo
	Processor map[component.Type]ModuleInfo
	Exporter  map[component.Type]ModuleInfo
	Extension map[component.Type]ModuleInfo
	Connector map[component.Type]ModuleInfo
}
```

This will allow to extend in the future if needed.

Reply:
Done, PTAL